src/Entity/ProductBundleItem.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace Drupal\commerce_product\Entity\ProductBundleItem;
+
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Field\BaseFieldDefinition;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Entity\ContentEntityInterface;
+
+/**
+ * Defines the bundle item entity.
+ *
+ * @ContentEntityType(
+ *   id = "bundle_item",
+ *   label = @Translation("bundle item"),
+ *   base_table = "commerce_product_bundle_item",
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "uuid" = "uuid",
+ *   },
+ * )
+ */
+
+class ProductBundleItem extends ContentEntityBase implements ContentEntityInterface {
+
+  public static function baseFieldDefinitions(EntityTypeInterface $entity_type) {
+    $fields['product_id'] = BaseFieldDefinition::create('entity_reference')
+      ->setLabel(t('Product Variation'))
+      ->setDescription(t('The included product variation.'))
+      ->setSetting('target_type', 'commerce_product_variation')
```

Comment:
this will work for now, but we'll need to figure out how to be able to reference any entity that implements PurchasableEntityInterface